wandb/env.py

```diff
@@ -81,6 +83,7 @@
 GIT_COMMIT = "WANDB_GIT_COMMIT"
 GIT_REMOTE_URL = "WANDB_GIT_REMOTE_URL"
 _EXECUTABLE = "WANDB_EXECUTABLE"
+ASYNC_UPLOAD_CONCURRENCY_LIMIT = "WANDB_ASYNC_UPLOAD_CONCURRENCY_LIMIT"
```

Comment:
also what is your though about making it part of `wandb.require` mechanism? maybe env is enough but we usually use it for new experimental features, but not sure if that makes a big difference, just bringing to your attention that we have this feature

Reply:
Hmm! I'd never heard of `wandb.require`! It looks like a pretty reasonable place to stick this, to my unpracticed eye.

IIUC, if we do `wandb.require`, then, instead of the user doing something like
```
WANDB_ASYNC_UPLOAD_CONCURRENCY_LIMIT=500 python3 do-ml-stuff.py
```
they would put a line in `do-ml-stuff.py` like
```python
wandb.require("async-upload")
```
and... I guess there would be some other API for them to configure the concurrency limit, if the sensible-default doesn't work for them.

Does that sound right to you?

If so: naively, the Settings approach seems a bit cleaner to me, just because it doesn't require two separate APIs (one for enable/disable, one for concurrency-control). Disagree?

Reply:
Yeah, your understanding is correct. I could see your point. just wanted to bring it your attention, since I assumed you didn't know about this feature.

Reply:
Indeed I did not! "How _should_ this be configured, given that env vars suck?" was one of the questions I'd forgotten I'd deferred, so thanks for bringing it up!